merlin-driver/src/main/java/gov/nasa/jpl/aerie/merlin/driver/engine/SimulationEngine.java

```diff
@@ -779,39 +804,44 @@ static ConditionJobId forCondition(final ConditionId condition) {
   }
 
   /** The lifecycle stages every task passes through. */
-  private sealed interface ExecutionState<Return> {
-    /** The task is in its primary operational phase. */
-    record InProgress<Return>(Duration startOffset, Task<Return> state)
-        implements ExecutionState<Return>
+  private sealed interface ExecutionState<Output> {
+    record InProgress<Input, Output>(
```

Comment:
Any thoughts on the pros/cons of an "InProgress" with a mutable input, vs an explicit "Blocked" state?

If we do go with the MutableObject - it might be helpful to spell out the invariants in a comment.

Taking a quick stab at the invariants I'm inferring - the contents of input will be immediately `Unit` for all tasks whose input is `Unit` - and for tasks with other types of input, the contents will be initially `null`, and then must be updated to the correct value before signaling the task?